crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -176,23 +183,50 @@ pub trait BuildChildren {
     fn add_child(&mut self, child: Entity) -> &mut Self;
 }
 
-impl<'w, 's, 'a> BuildChildren for EntityCommands<'w, 's, 'a> {
-    fn with_children(&mut self, spawn_children: impl FnOnce(&mut ChildBuilder)) -> &mut Self {
+/// Result of a [`BuildChildren::with_children`] call that provides access to the underlying commands
+/// and the closure's output.
+pub struct WithChildren<'w, 's, 'a, 'b, T> {
```

Comment:
I don't like this struct. The only cases where it's useful require putting it into a local variable or just losing the commands. I think if you're already putting it into a local variable (because you want both), it would be easy enough to just put the commands into the local variable one line higher up, as it were.

That is, I think if we want this method, it should have a signature more like `add_children<T>(&mut self, ...)->T`, keeping the current method as is.

Alternatively, we could just encourage people to use local variables and overwrite them in the `spawn_children`. One small issue with that is that they would need to use something like `Option::unwrap`.

Reply:
> Alternatively, we could just encourage people to use local variables and overwrite them in the spawn_children. One small issue with that is that they would need to use something like Option::unwrap.

Yeah, that's what I'm trying to avoid.

> That is, I think if we want this method, it should have a signature more like add_children<T>(&mut self, ...)->T, keeping the current method as is.

I like this idea, will do!